src/socket/dhcpv4.rs

```diff
@@ -151,7 +185,38 @@ impl Socket {
             config_changed: true,
             transaction_id: 1,
             max_lease_duration: None,
+            retry_config: &RetryConfig::DEFAULT,
             ignore_naks: false,
+            bootfile_name_buffer: None,
+            outbox_options: None,
+            inbox_options: None,
+            parameter_request_list: None,
+            #[cfg(feature = "async")]
+            waker: WakerRegistration::new(),
+        }
+    }
+
+    /// Create a DHCPv4 socket with retry and DHCP option configuration.
+    pub fn with_config(
```

Comment:
instead of an alternate constructor, can you add getters/setters for the configurable things? This is what all other sockets do, see TCP for example.

Reply:
Done.